src/Healthchecks.php

```diff
@@ -99,27 +99,31 @@ public function configureChecksOptions()
 
 
 	/**
-	 * @return Checks[]
+	 * @param array $tags To filter checks by their tags
+	 *
+	 * @return array
 	 */
-	public function listAllChecks(): array
+	public function listAllChecks(array $tags = []): array
 	{
 		$checksPerClient = array_map(
```

Comment:
please add `Assertion::allIsString($tags)` or similar at the very beginning

Reply:
Didn't do it like that, but every item in `$tags` array will be passed to `pairTagParam` method, which has a signature `string $tag`.
If an element is not a string it will raise a type error.